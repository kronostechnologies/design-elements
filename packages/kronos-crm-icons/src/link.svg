<?xml version="1.0" encoding="UTF-8"?>
<svg width="20px" height="20px" viewBox="0 0 20 20" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 49.2 (51160) - http://www.bohemiancoding.com/sketch -->
    <title>link</title>
    <desc>Created with Sketch.</desc>
    <defs></defs>
    <g id="link" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g fill="#000000" fill-rule="nonzero" id="Shape">
            <path d="M7.32375,15.8659218 L6.31125,16.8721718 C5.435,17.7409218 4.0075,17.7409218 3.13,16.8721718 C2.71,16.4546718 2.47875,15.8984218 2.47875,15.3071718 C2.47875,14.7159218 2.71125,14.1621718 3.13,13.7421718 L6.855,10.0484218 C7.62625,9.28217183 9.07875,8.15467183 10.1375,9.20467183 C10.62375,9.68717183 11.4075,9.68467183 11.89125,9.19842183 C12.3725,8.71217183 12.37,7.92717183 11.88375,7.44592183 C10.08625,5.66092183 7.42625,5.99092183 5.11,8.28967183 L1.385,11.9846718 C0.49125,12.8709218 0,14.0521718 0,15.3071718 C0,16.5646718 0.4925,17.7434218 1.385,18.6296718 C2.305,19.5421718 3.5125,19.9996718 4.72125,19.9996718 C5.93,19.9996718 7.13875,19.5434218 8.0575,18.6296718 L9.07125,17.6234218 C9.5575,17.1421718 9.56,16.3584218 9.0775,15.8734218 C8.595,15.3884218 7.81,15.3859218 7.32375,15.8659218 Z M18.61375,1.51092183 C16.68,-0.406578175 13.9775,-0.510328175 12.19,1.26467183 L10.92875,2.51717183 C10.4425,2.99967183 10.43875,3.78342183 10.92125,4.26842183 C11.40375,4.75467183 12.1875,4.75717183 12.67375,4.27467183 L13.93625,3.02342183 C14.86125,2.10342183 16.075,2.48467183 16.86875,3.26967183 C17.28875,3.68842183 17.52125,4.24342183 17.52125,4.83467183 C17.52125,5.42592183 17.28875,5.98092183 16.86875,6.39842183 L12.89375,10.3409218 C11.07625,12.1421718 10.22375,11.2984218 9.86,10.9371718 C9.37375,10.4546718 8.59,10.4584218 8.10875,10.9434218 C7.6275,11.4284218 7.62875,12.2146718 8.115,12.6946718 C8.95,13.5221718 9.9025,13.9321718 10.9,13.9321718 C12.12125,13.9321718 13.4125,13.3171718 14.64125,12.0984218 L18.61625,8.15717183 C19.50625,7.26967183 20,6.08967183 20,4.83467183 C20,3.57967183 19.50625,2.39842183 18.61375,1.51092183 Z"></path>
        </g>
    </g>
</svg>