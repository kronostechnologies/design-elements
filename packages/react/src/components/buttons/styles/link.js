export default `
  align-items: center;
  display: inline-flex;
  text-decoration: none;
`;
