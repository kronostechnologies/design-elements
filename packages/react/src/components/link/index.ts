export { Link } from './link';
export type { LinkButtonProps, LinkProps } from './types';
