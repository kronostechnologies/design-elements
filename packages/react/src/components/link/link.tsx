import { FC, ReactElement, MouseEvent } from 'react';
import { useId } from '../../hooks/use-id';
import { useTranslation } from '../../i18n/use-translation';
import { useDeviceContext } from '../device-context-provider/device-context-provider';
import { Icon } from '../icon/icon';
import { ScreenReaderOnlyText } from '../screen-reader-only-text/ScreenReaderOnlyText';
import { Tooltip } from '../tooltip/tooltip';
import { StyledLink } from './styled';
import { LinkProps } from './types';

export const Link: FC<LinkProps> = ({
    button,
    children,
    disabled = false,
    external = false,
    href = '#',
    icon,
    id: providedId,
    target = '_blank',
    onClick,
    routerLink,
    rel = 'noopener noreferrer',
    ...linkProps
}): ReactElement => {
    const id = useId(providedId);
    const { isMobile } = useDeviceContext();
    const { t } = useTranslation('common');
    const opensInNewTab = external && target === '_blank';
    const isIconOnly = icon && !children;

    const handleClick = (event: MouseEvent<HTMLAnchorElement>): void => {
        if (disabled) {
            event.preventDefault();
            event.stopPropagation();
        } else {
            onClick?.(event);
        }
    };

    const renderLink = (): ReactElement => (
        <StyledLink
            as={external ? 'a' : routerLink}
            to={external ? undefined : href}
            data-testid="link"
            aria-disabled={disabled ? 'true' : 'false'}
            aria-label={isIconOnly ? icon.label : undefined}
            id={id}
            href={disabled ? undefined : href}
            onClick={handleClick}
            target={external ? target : undefined}
            tabIndex={disabled ? -1 : 0}
            rel={external ? rel : undefined}
            $buttonProps={button}
            $isMobile={isMobile}
            $disabled={disabled}
            {...linkProps /* eslint-disable-line react/jsx-props-no-spreading */}
        >
            {icon?.name && (
                <Icon
                    data-testid="link-icon"
                    aria-hidden="true"
                    name={icon.name}
                    size="16"
                />
            )}
            {!isIconOnly && children}
            {external && (
                <Icon
                    data-testid="external-link-icon"
                    aria-hidden="true"
                    focusable={false}
                    name="externalLink"
                    role="img"
                    size="16"
                />
            )}
            {opensInNewTab && (
                <ScreenReaderOnlyText
                    data-testid="screen-reader-text"
                    label={t('opensInNewTabScreenReader')}
                />
            )}
        </StyledLink>
    );

    return isIconOnly ? (
        <Tooltip
            aria-describedby={`${id}-tooltip`}
            label={icon.label}
        >
            {renderLink()}
        </Tooltip>
    ) : renderLink();
};

Link.displayName = 'Link';
