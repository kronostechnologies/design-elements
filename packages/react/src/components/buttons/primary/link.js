import { Link } from 'react-router-dom';
import styled from 'styled-components';

import abstract from '../styles/abstract';
import primary from '../styles/primary';
import link from '../styles/link';

export default styled(Link)`
  ${abstract}
  ${primary}
  ${link}
`;
