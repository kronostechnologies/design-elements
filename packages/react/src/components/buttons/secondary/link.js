import { Link } from 'react-router-dom';
import styled from 'styled-components';

import abstract from '../styles/abstract';
import secondary from '../styles/secondary';
import link from '../styles/link';

export default styled(Link)`
  ${abstract}
  ${secondary}
  ${link}
`;
