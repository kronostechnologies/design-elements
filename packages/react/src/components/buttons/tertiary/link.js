import { Link } from 'react-router-dom';
import styled from 'styled-components';

import abstract from '../styles/abstract';
import tertiary from '../styles/tertiary';
import link from '../styles/link';

export default styled(Link)`
  ${abstract}
  ${tertiary}
  ${link}
`;
