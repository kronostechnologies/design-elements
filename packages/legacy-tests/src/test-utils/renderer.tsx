import { CommonWrapper, mount, MountRendererProps, ReactWrapper, render } from 'enzyme';
import { act, Component, FunctionComponent, PropsWithChildren, ReactElement } from 'react';
import { MemoryRouter } from 'react-router-dom';
import { DesignSystem, DesignSystemProps } from '~/components/design-system';
import { ThemeWrapper } from '~/components/theme-wrapper/theme-wrapper';

export const AllProviders: FunctionComponent<PropsWithChildren<DesignSystemProps>> = ({
    children,
    language,
    theme,
    staticDevice,
}) => (
    <MemoryRouter future={{ v7_startTransition: false, v7_relativeSplatPath: false }}>
        <DesignSystem language={language} staticDevice={staticDevice} theme={theme}>
            {children}
        </DesignSystem>
    </MemoryRouter>
);

interface WrappingComponentProps {
    wrappingComponentProps?: DesignSystemProps;
}

type Options = MountRendererProps & WrappingComponentProps;

export function mountWithProviders<C extends Component, P = C['props'], S = C['state']>(
    component: ReactElement<P>,
    options: Options = {},
): ReactWrapper<P, S, C> {
    return mount(component, {
        ...options,
        wrappingComponent: AllProviders as Options['wrappingComponent'],
    });
}

export function mountWithTheme<C extends Component, P = C['props'], S = C['state']>(
    component: ReactElement<P>,
    options?: MountRendererProps,
): ReactWrapper<P, S, C> {
    return mount(component, {
        wrappingComponent: ThemeWrapper as MountRendererProps['wrappingComponent'],
        ...options,
    });
}

export function renderWithTheme(
    component: ReactElement,
): cheerio.Cheerio {
    return render(
        <ThemeWrapper>
            {component}
        </ThemeWrapper>,
    );
}

export async function actUpdate<C extends Component, P = C['props'], S = C['state']>(
    wrapper: CommonWrapper<P, S, C>,
): Promise<void> {
    await act(async () => {
        wrapper.update();
    });
}

export async function actAndWaitForEffects<C extends Component, P = C['props'], S = C['state']>(
    wrapper: ReactWrapper<P, S, C>,
    action: () => void,
): Promise<ReactWrapper<P, S, C>> {
    await act(async () => action());
    await actUpdate(wrapper);
    return wrapper;
}
