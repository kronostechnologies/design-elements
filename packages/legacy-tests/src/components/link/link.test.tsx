import { act } from 'react-dom/test-utils';
import { Link as RouteLink, NavLink } from 'react-router-dom';
import { Link } from '~/components/link/link';
import { mountWithProviders } from '../../test-utils/renderer';

describe('Link Component', () => {
    describe('Features', () => {
        it('internal link is valid and updates dynamically ', () => {
            const wrapper = mountWithProviders(<Link href="/internal-page" />);

            expect(wrapper.find('a').prop('href')).toBe('/internal-page');
        });

        it('external link is valid', () => {
            const wrapper = mountWithProviders(<Link external href="https://example.com" />);

            expect(wrapper.find('a').prop('href')).toBe('https://example.com');
            expect(wrapper.find('a').prop('target')).toBe('_blank');
            expect(wrapper.find('a').prop('rel')).toContain('noopener noreferrer');
        });

        it('renders icon', () => {
            const wrapper = mountWithProviders(<Link icon={{ name: 'mail', label: 'This is a label' }}>Label</Link>);

            expect(wrapper.find('Icon').prop('name')).toBe('mail');
        });

        it('renders tooltip isIconOnly is provided', () => {
            const tooltipWrapper = mountWithProviders(<Link icon={{ name: 'mail', label: 'Test Label' }} />);

            expect(tooltipWrapper.find('Tooltip').prop('label')).toBe('Test Label');
            expect(tooltipWrapper.find('Icon').exists()).toBe(true);
        });

        it('renders children when provided', () => {
            const childrenWrapper = mountWithProviders(<Link><div className="test-child">Child Content</div></Link>);

            expect(childrenWrapper.contains(<div className="test-child">Child Content</div>)).toBe(true);
        });

        it('renders NavLink when routerLink prop is NavLink', () => {
            const wrapper = mountWithProviders(<Link routerLink={NavLink} href="/internal-page" />);

            expect(wrapper.find(NavLink).exists()).toBe(true);
            expect(wrapper.find(NavLink).prop('to')).toBe('/internal-page');
        });

        it('renders RouteLink when routerLink prop is RouteLink', () => {
            const routeLinkWrapper = mountWithProviders(<Link routerLink={RouteLink} href="/internal-route" />);

            expect(routeLinkWrapper.find(RouteLink).exists()).toBe(true);
            expect(routeLinkWrapper.find(NavLink).exists()).toBe(false);
            expect(routeLinkWrapper.find(RouteLink).prop('to')).toBe('/internal-route');
        });

        it('calls onClick callback when clicked', () => {
            const onClickMock = jest.fn();
            const wrapper = mountWithProviders(<Link href="/test" onClick={onClickMock} />);

            wrapper.find('a').simulate('click');

            expect(wrapper.find('a').prop('href')).toBeDefined();
            expect(onClickMock).toHaveBeenCalledTimes(1);
        });

        it('prevent onClick callback when disabled', () => {
            const onClickMock = jest.fn();
            const wrapper = mountWithProviders(<Link href="/test" onClick={onClickMock} />);

            wrapper.find('a').simulate('click');
            act(() => {
                wrapper.setProps({ disabled: true });
            });
            wrapper.find('a').simulate('click');

            expect(wrapper.find('a').prop('href')).toBeUndefined();
            expect(onClickMock).toHaveBeenCalledTimes(1);
        });

        it('updates dynamically ', () => {
            const wrapper = mountWithProviders(<Link href="/internal-page" />);

            act(() => {
                wrapper.setProps({ href: '/updated-page' });
                wrapper.setProps({ icon: { name: 'bell', label: 'This is a label' } });
            });

            expect(wrapper.find('a').prop('href')).toBe('/updated-page');
            expect(wrapper.find('Icon').prop('name')).toBe('bell');
        });
    });

    describe('Styling', () => {
        it('matches default snapshot', () => {
            const tree = mountWithProviders(<Link />);

            expect(tree).toMatchSnapshot();
        });

        it('matches with children snapshot', () => {
            const tree = mountWithProviders(
                <Link><span>Test</span></Link>,
            );

            expect(tree).toMatchSnapshot();
        });

        it('matches NavLink snapshot', () => {
            const tree = mountWithProviders(
                <Link routerLink={NavLink} href="/test">Navigation Link</Link>,
            );

            expect(tree).toMatchSnapshot();
        });

        it('matches RouteLink snapshot', () => {
            const tree = mountWithProviders(
                <Link routerLink={RouteLink} href="/test">Navigation Link</Link>,
            );

            expect(tree).toMatchSnapshot();
        });
    });

    describe('Accessibility', () => {
        it('applies aria-disabled when link is disabled', () => {
            const wrapper = mountWithProviders(<Link disabled />);

            expect(wrapper.find('a').prop('aria-disabled')).toBe('true');
        });

        it('displays screen-reader-only text only when external && target="_blank"', () => {
            const wrapper = mountWithProviders(<Link external target='none' />);

            expect(wrapper.find('ScreenReaderOnlyText').exists()).toBe(false);

            act(() => {
                wrapper.setProps({ external: true, target: '_blank' });
            });

            expect(wrapper.find('ScreenReaderOnlyText').exists()).toBe(true);
            expect(wrapper.find('ScreenReaderOnlyText').text()).toContain('opens in a new tab');
        });
    });
});
