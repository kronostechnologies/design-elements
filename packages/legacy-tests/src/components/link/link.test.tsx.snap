// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Component Styling matches NavLink snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: var(--border-radius);
  color: #006296;
  cursor: pointer;
  font-size: 0.875rem;
  font-style: normal;
  font-weight: var(--font-normal);
  -webkit-letter-spacing: 0.0125rem;
  -moz-letter-spacing: 0.0125rem;
  -ms-letter-spacing: 0.0125rem;
  letter-spacing: 0.0125rem;
  line-height: 1.25rem;
  padding: 0 var(--spacing-half);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: var(--spacing-half);
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 svg {
  color: #006296;
  height: 1rem;
  width: 1rem;
}

.c0[visited],
.c0:visited {
  color: #006296;
}

.c0[visited] svg,
.c0:visited svg {
  color: #006296;
}

.c0[hover],
.c0:hover {
  color: #003A5A;
}

.c0[hover] svg,
.c0:hover svg {
  color: #003A5A;
}

.c0[disabled],
.c0[aria-disabled='true'],
.c0:disabled {
  cursor: not-allowed;
  pointer-events: none;
}

.c0 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c0:focus-visible {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

<Link
  href="/test"
  routerLink={
    {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
>
  <styled.a
    $disabled={false}
    $isMobile={false}
    aria-disabled="false"
    as={
      {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
    data-testid="link"
    href="/test"
    id="uuid1"
    onClick={[Function]}
    tabIndex={0}
    to="/test"
  >
    <NavLink
      aria-disabled="false"
      className="c0"
      data-testid="link"
      href="/test"
      id="uuid1"
      onClick={[Function]}
      tabIndex={0}
      to="/test"
    >
      <Link
        aria-disabled="false"
        className="c0"
        data-testid="link"
        href="/test"
        id="uuid1"
        onClick={[Function]}
        tabIndex={0}
        to="/test"
      >
        <a
          aria-disabled="false"
          className="c0"
          data-testid="link"
          href="/test"
          id="uuid1"
          onClick={[Function]}
          tabIndex={0}
        >
          Navigation Link
        </a>
      </Link>
    </NavLink>
  </styled.a>
</Link>
`;

exports[`Link Component Styling matches RouteLink snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: var(--border-radius);
  color: #006296;
  cursor: pointer;
  font-size: 0.875rem;
  font-style: normal;
  font-weight: var(--font-normal);
  -webkit-letter-spacing: 0.0125rem;
  -moz-letter-spacing: 0.0125rem;
  -ms-letter-spacing: 0.0125rem;
  letter-spacing: 0.0125rem;
  line-height: 1.25rem;
  padding: 0 var(--spacing-half);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: var(--spacing-half);
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 svg {
  color: #006296;
  height: 1rem;
  width: 1rem;
}

.c0[visited],
.c0:visited {
  color: #006296;
}

.c0[visited] svg,
.c0:visited svg {
  color: #006296;
}

.c0[hover],
.c0:hover {
  color: #003A5A;
}

.c0[hover] svg,
.c0:hover svg {
  color: #003A5A;
}

.c0[disabled],
.c0[aria-disabled='true'],
.c0:disabled {
  cursor: not-allowed;
  pointer-events: none;
}

.c0 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c0:focus-visible {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

<Link
  href="/test"
  routerLink={
    {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
>
  <styled.a
    $disabled={false}
    $isMobile={false}
    aria-disabled="false"
    as={
      {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
    data-testid="link"
    href="/test"
    id="uuid1"
    onClick={[Function]}
    tabIndex={0}
    to="/test"
  >
    <Link
      aria-disabled="false"
      className="c0"
      data-testid="link"
      href="/test"
      id="uuid1"
      onClick={[Function]}
      tabIndex={0}
      to="/test"
    >
      <a
        aria-disabled="false"
        className="c0"
        data-testid="link"
        href="/test"
        id="uuid1"
        onClick={[Function]}
        tabIndex={0}
      >
        Navigation Link
      </a>
    </Link>
  </styled.a>
</Link>
`;

exports[`Link Component Styling matches default snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: var(--border-radius);
  color: #006296;
  cursor: pointer;
  font-size: 0.875rem;
  font-style: normal;
  font-weight: var(--font-normal);
  -webkit-letter-spacing: 0.0125rem;
  -moz-letter-spacing: 0.0125rem;
  -ms-letter-spacing: 0.0125rem;
  letter-spacing: 0.0125rem;
  line-height: 1.25rem;
  padding: 0 var(--spacing-half);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: var(--spacing-half);
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 svg {
  color: #006296;
  height: 1rem;
  width: 1rem;
}

.c0[visited],
.c0:visited {
  color: #006296;
}

.c0[visited] svg,
.c0:visited svg {
  color: #006296;
}

.c0[hover],
.c0:hover {
  color: #003A5A;
}

.c0[hover] svg,
.c0:hover svg {
  color: #003A5A;
}

.c0[disabled],
.c0[aria-disabled='true'],
.c0:disabled {
  cursor: not-allowed;
  pointer-events: none;
}

.c0 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c0:focus-visible {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

<Link>
  <styled.a
    $disabled={false}
    $isMobile={false}
    aria-disabled="false"
    data-testid="link"
    href="#"
    id="uuid1"
    onClick={[Function]}
    tabIndex={0}
    to="#"
  >
    <a
      aria-disabled="false"
      className="c0"
      data-testid="link"
      href="#"
      id="uuid1"
      onClick={[Function]}
      tabIndex={0}
      to="#"
    />
  </styled.a>
</Link>
`;

exports[`Link Component Styling matches with children snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: var(--border-radius);
  color: #006296;
  cursor: pointer;
  font-size: 0.875rem;
  font-style: normal;
  font-weight: var(--font-normal);
  -webkit-letter-spacing: 0.0125rem;
  -moz-letter-spacing: 0.0125rem;
  -ms-letter-spacing: 0.0125rem;
  letter-spacing: 0.0125rem;
  line-height: 1.25rem;
  padding: 0 var(--spacing-half);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: var(--spacing-half);
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 svg {
  color: #006296;
  height: 1rem;
  width: 1rem;
}

.c0[visited],
.c0:visited {
  color: #006296;
}

.c0[visited] svg,
.c0:visited svg {
  color: #006296;
}

.c0[hover],
.c0:hover {
  color: #003A5A;
}

.c0[hover] svg,
.c0:hover svg {
  color: #003A5A;
}

.c0[disabled],
.c0[aria-disabled='true'],
.c0:disabled {
  cursor: not-allowed;
  pointer-events: none;
}

.c0 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c0:focus-visible {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

<Link>
  <styled.a
    $disabled={false}
    $isMobile={false}
    aria-disabled="false"
    data-testid="link"
    href="#"
    id="uuid1"
    onClick={[Function]}
    tabIndex={0}
    to="#"
  >
    <a
      aria-disabled="false"
      className="c0"
      data-testid="link"
      href="#"
      id="uuid1"
      onClick={[Function]}
      tabIndex={0}
      to="#"
    >
      <span>
        Test
      </span>
    </a>
  </styled.a>
</Link>
`;
